specs/Standards/ContractMetadata.md

```diff
@@ -0,0 +1,43 @@
+# Contract Metadata
+
+## [NEP-330](https://github.com/near/NEPs/discussions/329)
+
+Version `1.0.0`
+
+## Summary
+
+The contract metadata is a standard interface that allows auditing and viewing source code for a deployed smart contract. Implementation of this standard is purely optional but is recommended for developers whose contracts are open source.
+
+## Motivation
+
+There is no trivial way of finding the source code or author of a deployed smart contract. Having a standard that outlines how to view the source code of an arbitrary smart contract creates an environment of openness and collaboration.
+
+The initial discussion can be found [here](https://github.com/near/NEPs/discussions/329).
+
+## Interface
+
+Metadata applies at the contract level (`ContractMetadata`):
+
+```ts
+type ContractMetadata = {
+  version: string|null, // optional, commit hash being used for the currently deployed wasm. If the contract is not open-sourced, this could also be a numbering system for internal organization / tracking such as "1.0.0" and "2.1.0".
+  link: string|null, //optional,  link to open source code such as a Github repository or a CID to somewhere on IPFS.
```

Comment:
What about adding some sort of organization contact link as well that points to the website of the application, or the developers; potentially with more contact information!

The difference between it and `link` is that one is to discover the source code, and the other is to discover what is the preferred way to reach developers/community behind this smart contract.

Reply:
I think this information could be found by utilizing the `link` since you can open issues or PRs on Github and follow or view the developers' public profiles. The case where I think this might be useful, is if the link is a CID to somewhere on IPFS since you don't have the same ability to reach out or view the developers as you do using Github.

thoughts? @austinabell @agileurbanite

Reply:
I would actually go even further and provide three different fields (similar to [crates.io](https://crates.io/crates/rand)): `Homepage`, `Documentation` and `Repository` + `version` (to pin the exact version and allow reproducing the exact same binary).

Some services like [npm](https://www.npmjs.com/package/near-api-js) only have `Homepage` and `Repository`. (Maybe `documentation` is a bit too much).